ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt

```diff
@@ -92,17 +92,16 @@ internal fun CoroutineScope.attachForReadingDirectImpl(
         channel.writeSuspendSession {
             while (true) {
                 var rc = 0
+                val buffer = request(1)
+                if (buffer == null) {
+                    if (channel.isClosedForWrite) break
```

Comment:
I think we should do the same for `isClosedForWrite`. We should not lose exceptions in writing.

Reply:
Sure, will fix